### PR TITLE
[pose-detection] Setup demo page[2]

### PR DESCRIPTION
**Pull request by lina128**
This PR include below changes:
1. Setup camera, so that it keeps a time state for the video stream, manages video lifecycle and canvas drawing.
2. Setup stats for fps.
3. Setup gui for option tuning. Added tunable options for camera. To add more model related later.
4. Fixed a bug during migration in PoseNet keypoint projection.
5. Added keypoints drawing. To add more rendering later, e.g. drawing based on confidence levels, skeleton and box rendering, annotation/color coding, etc.

p.s. Tested with PoseNet single pose.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)

### CODE REVIEW

**Review by pyu10055 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)* status: 0 of 1 approvals obtained (waiting on @lina128 and @mhays-google)

___
*[pose-detection/demo/src/index.js, line 34 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/615#-MWC7jg5BCk-BFZe5hKE:-MWC7jg5BCk-BFZe5hKF:bobjm25) ([raw file](https://github.com/tensorflow/tfjs-models/blob/67695f5180bf4cb62b7994b569ddf52d89158250/pose-detection/demo/src/index.js#L34)):*
> ```JavaScript
>     STATE.camera.targetFPS = STATE.changeToTargetFPS;
>     STATE.changeToTargetFPS = null;
>     camera = await Camera.setupCamera(STATE.camera);
> ```

should the setupCamera be called only once even both changeToTargetFPS and changeToSizeOption are set?
___
*[pose-detection/demo/src/index.js, line 57 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/615#-MWC8Cc83lKnzty4LM42:-MWC8Cc83lKnzty4LM43:bqrei78) ([raw file](https://github.com/tensorflow/tfjs-models/blob/67695f5180bf4cb62b7994b569ddf52d89158250/pose-detection/demo/src/index.js#L57)):*
> ```JavaScript
>       camera.drawResult(poses[0]);
>     } else {
>       camera.clearCtx();
> ```

should clearCtx() always be called regardless of poses detected?

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)* status: 0 of 1 approvals obtained (waiting on @ardoerlemans, @mhays-google, and @pyu10055)

___
*[pose-detection/demo/src/index.js, line 34 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/615#-MWC7jg5BCk-BFZe5hKE:-MWPyOpV6oSvuT_0ASoI:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/67695f5180bf4cb62b7994b569ddf52d89158250/pose-detection/demo/src/index.js#L34)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

should the setupCamera be called only once even both changeToTargetFPS and changeToSizeOption are set?
</blockquote></details>

Done.
___
*[pose-detection/demo/src/index.js, line 57 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/615#-MWC8Cc83lKnzty4LM42:-MWPzejmBxIgKhXeVovN:b-zieck) ([raw file](https://github.com/tensorflow/tfjs-models/blob/67695f5180bf4cb62b7994b569ddf52d89158250/pose-detection/demo/src/index.js#L57)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

should clearCtx() always be called regardless of poses detected?
</blockquote></details>

clearCtx will remove the result from drawImage as well, so removed.

**Review by pyu10055 — APPROVED** [review]
> Reviewed 9 of 10 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ardoerlemans, @mhays-google, and @pyu10055)

File: pose-detection/demo/src/params.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+export const DEFAULT_LINE_WIDTH = 4;
+
+export const VIDEO_SIZE = {
+  '640 X 480': {width: 640, height: 480},
+  '500 X 500': {width: 500, height: 500}
```
Comment (ardoerlemans) [inline pose-detection/demo/src/params.js:21]: I don't think cameras will support this resolution.

**Review by ardoerlemans — APPROVED** [review]

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans, @mhays-google, and @pyu10055)

___
*[pose-detection/demo/src/params.js, line 21 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/615#-MWC0UrAVoSa_1MZXjjw-r2-21:-MWQyduL62Aohn1l2vPA:b-144bc2) ([raw file](https://github.com/tensorflow/tfjs-models/blob/91458ce677730007456a0422c6dd8da29fc208e6/pose-detection/demo/src/params.js#L21)):*
<details><summary><i>Previously, ardoerlemans wrote…</i></summary><blockquote>

I don't think cameras will support this resolution.
</blockquote></details>

Removed the option. Added 640 x 360. Thank you for the review!

**Review by mhays-google — COMMENTED** [review]
> Reviewed 8 of 10 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans and @pyu10055)

**Review by mhays-google — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans and @pyu10055)

___
*[pose-detection/demo/src/params.js, line 21 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/615#-MWC0UrAVoSa_1MZXjjw-r2-21:-MWRc2c45TsJ7FbFuvml:b9bggjy) ([raw file](https://github.com/tensorflow/tfjs-models/blob/91458ce677730007456a0422c6dd8da29fc208e6/pose-detection/demo/src/params.js#L21)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Removed the option. Added 640 x 360. Thank you for the review!
</blockquote></details>

I seem to remember that you can plug anything you want into this and it will find the closest match.  It might not be terrible to allow people to specify `{h:number, w:number}`

**Review by mhays-google — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/615)* status: :shipit: complete! 3 of 1 approvals obtained (waiting on @ardoerlemans and @pyu10055)

**Pull request merged**